js/components/productList.js

```diff
@@ -1,4 +1,96 @@
-import fetchProducts from "../fetch/fetch.js";
+
+import fetchProducts, { fetchOneProduct } from "../fetch/fetch.js";
+import { deleteProduct } from "../fetch/deleteProduct.js";
+import { addProduct } from "../fetch/addProduct.js";
+import { editProduct } from "../fetch/editProduct.js";
+
+const state = {
+  activeProductID: "",
+
+  activateModal(id) {
+    document.getElementById(id).classList.add("modal-active");
+  },
+  deactivateModal() {
+    document.querySelector(".modal-active").classList.remove("modal-active");
+    this.activeProductID = "";
+  },
+  createProductObject(formID) {
+    const dataObj = {};
+    //using form data as it ignores checkbox value and checks if it's checked, entries list name-value pairs
+    const formData = new FormData(document.getElementById(formID));
+    const entriesIterator = formData.entries();
+
+    for (let i of entriesIterator) {
+      //if has a value (checkbox is ignored if not checked)
+      if (i[1]) {
+        dataObj[i[0]] = i[1];
+      }
+    }
+
+    return dataObj;
+  },
+  addNewProduct(product) {
```

Comment:
Since we do not have types in JS, consider renaming method parameters. Maybe `productElement`